user/src/com/google/gwt/dom/client/DOMImpl.java

```diff
@@ -138,7 +138,9 @@ public native boolean eventGetMetaKey(NativeEvent evt) /*-{
     return !!evt.metaKey;
   }-*/;
 
-  public abstract int eventGetMouseWheelVelocityY(NativeEvent evt);
+  public int eventGetMouseWheelVelocityY(NativeEvent evt) {
+    return evt.getDeltaY() == 0.0 ? 0 : (evt.getDeltaY() > 0 ? 1 : -1);
```

Comment:
That works, but I was thinking `return (int) Math.signum(evt.getDeltaY());` - at worst it'll do the same thing you're doing, at best it'll share the same pair of conditionals with other code using that impl.

This isn't meant to be a criticism, as I might not have thought this through fully, just pointing it out in case there's a path here that makes sense.